shared/src/main/java/gov/hhs/cdc/trustedintermediary/external/HikariConnectionPool.java

```diff
@@ -0,0 +1,61 @@
+package gov.hhs.cdc.trustedintermediary.external;
+
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import gov.hhs.cdc.trustedintermediary.context.ApplicationContext;
+import gov.hhs.cdc.trustedintermediary.wrappers.database.ConnectionPool;
+import gov.hhs.cdc.trustedintermediary.wrappers.database.DatabaseCredentialsProvider;
+import java.sql.Connection;
+import java.sql.SQLException;
+
+/**
+ * A shared connection pool for connecting to the database. The class currently spins up 10
+ * connections on server startup. This is the default Hikari behavior. For troubleshooting or tuning
+ * help please check the HikariCP repo for info on how to tune the connection performance should any
+ * issues arise.
+ */
+public class HikariConnectionPool implements ConnectionPool {
+
+    private static HikariConnectionPool INSTANCE;
+
+    public final HikariDataSource ds;
+
+    private HikariConnectionPool() {
+        HikariConfig config = constructHikariConfig();
+        ds = new HikariDataSource(config);
+    }
+
+    public static HikariConfig constructHikariConfig() {
```

Comment:
Let's make this package private since we call it in the test.  Probably shouldn't be public.

```suggestion
    static HikariConfig constructHikariConfig() {
```